pkg/utils/wrappers.go

```diff
@@ -20,19 +20,20 @@ func DoWithRetry(
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
+	var err error
 	for {
 		select {
 		case <-ctx.Done():
-			return fmt.Errorf("retry timeout [%v] exceeded", timeout)
+			return fmt.Errorf("retry timeout [%v] exceeded: [%v]", timeout, err)
```

Comment:
In some cases, the err will be nil and in the other, it will be an error returned on the last retry. I'm not sure about including this error in the timeout error message as it's not strictly the error that caused the failure.
Maybe we should leave logging the intermediate errors to the `doFn` function implementation?

Reply:
The motivation here was that `utils.DoWithDefaultRetry` gobbles up useful error message, which breaks the `IsAddressUnused` tests in electrs_test.go.

What if I changed the verbiage to "most recent error: [%v]"? 

That way, if it was a timeout on the first attempt, we see: 
```
retry timeout [2m0s] exceeded. most recent error: [nil]
```
and otherwise we see something like
```
retry timeout [2m0s] exceeded. most recent error: [json: cannot unmarshal number into Go value of type []interface {}]
```

Reply:
Ok, let's replace the error message to: `"retry timeout [%v] exceeded; most recent error: [%v]"`